nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/test/java/org/apache/nifi/processors/kudu/MockScanKudu.java

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.kudu;
+
+import org.apache.kudu.ColumnSchema;
+import org.apache.kudu.Schema;
+import org.apache.kudu.Type;
+import org.apache.kudu.client.KuduTable;
+import org.apache.kudu.client.RowResult;
+import org.apache.kudu.client.RowResultIterator;
+import org.apache.kudu.client.Insert;
+import org.apache.nifi.processors.kudu.io.ResultHandler;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class MockScanKudu extends AbstractMockKuduProcessor {
+
+    private LinkedList<Insert> insertQueue;
+    private int numScans = 0;
+    private boolean throwException = false;
+    private int linesBeforeException = -1;
+    RowResultIterator testRows = mock(RowResultIterator.class);
+    List<RowResult> rowResultList = new ArrayList<>();
+
+    public MockScanKudu() {
+        this.insertQueue = new LinkedList<>();
+    }
+
+    public void queue(Insert... operations) {
+        insertQueue.addAll(Arrays.asList(operations));
+    }
+
+    @Override
+    public void scan(ProcessContext context, ProcessSession session, KuduTable kuduTable, String predicates, List<String> projectedColumnNames, ResultHandler handler) throws IOException {
+        if (throwException) {
+            throw new IOException("exception");
+        }
+
+        List<RowResult> matchedRows = new ArrayList<>();
+        Iterator<RowResult> it = testRows.iterator();
+        if (it != null) {
+            if (predicates == null || predicates.isEmpty() || !predicates.contains("=")) {
+                while (it.hasNext()) {
+                    matchedRows.add(it.next());
+                    handler.handle(it.next());
+                }
+            } else {
+                if (predicates.contains("=")) {
+                    final String[] parts = predicates.split("=");
+                    int i = 0;
+                    while (it.hasNext()) {
+                        if (linesBeforeException >= 0 && i++ >= linesBeforeException) {
+                            throw new IOException("iterating exception");
+                        }
+                        RowResult result = it.next();
+                        if (parts[1].equals(result.getString(String.valueOf(parts[0])))) {
+                            matchedRows.add(result);
+                            handler.handle(result);
+                        }
+                    }
+                }
+            }
+        }
+        numScans++;
+    }
+
+    public void addResult(Map<String, String> rows) {
+        rows.entrySet().forEach(kv -> rowResultList.add(createRowResult(kv.getKey(), kv.getValue())));
```

Comment:
Could be simplified to `rows.forEach((key, value) -> rowResultList.add(createRowResult(key, value)));`